module/registry/util/src/main/java/org/openbase/bco/registry/print/BCOQueryLabelPrinter.kt

```diff
@@ -0,0 +1,248 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * #L%
+ */ /**
+ * @author [Divine Threepwood](mailto:divine@openbase.org)
+ */
+object BCOQueryLabelPrinter {
+    const val APP_NAME = "bco-id-to-label"
+    @JvmStatic
+    fun main(args: Array<String>) {
+        var resultsFound = false
+        val query: String = if (args.isEmpty()) {
+            ""
+        } else {
+            args[args.size - 1]
+        }
+        try {
+
+            // handle jps
+            if (args.isNotEmpty()) {
+                // remove query from jp args
+                val jpArgs = args.copyOfRange(0, args.size - 1)
+                JPService.setApplicationName(APP_NAME)
+
+                // help
+                if (args.isNotEmpty() && (args[0] == "-h" || args[0] == "--help")) {
+                    JPService.printHelp()
+                    printHelp()
+                    exitProcess(0)
+                }
+                JPService.parseAndExitOnError(jpArgs)
+            }
+
+            // init
+            Registries.waitForData()
+            BCOLogin.getSession().autoLogin(true).get()
+            if (args.isEmpty()) {
+                println("At least one id or alias has to be provided!")
+                printHelp()
+                exitProcess(255)
+            }
+
+            // print by id
+            try {
+                printUnit(Registries.getUnitRegistry().getUnitConfigById(query))
+                resultsFound = true
+            } catch (ex: NotAvailableException) {
+                // continue if not available
+            }
+
+            // print by alias
+            try {
+                printUnit(Registries.getUnitRegistry().getUnitConfigByAlias(query))
+                resultsFound = true
+            } catch (ex: NotAvailableException) {
+                // continue if not available
+            }
+        } catch (ex: InterruptedException) {
+            println("killed")
+            exitProcess(253)
+        } catch (ex: Exception) {
+            ExceptionPrinter.printHistory(CouldNotPerformException("Could not query!", ex), System.err)
+            printHelp()
+            exitProcess(254)
+        }
+        if (resultsFound) {
+            exitProcess(0)
+        } else {
+            println(ConsoleColors.CYAN + "No match for " + ConsoleColors.RESET + query + ConsoleColors.CYAN + " found" + ConsoleColors.RESET + " :(")
+            printHelp()
+        }
+        exitProcess(255)
+    }
+
+    private fun printHelp() {
+        println()
+        println("Usage:     " + APP_NAME + " [options] -- [UNIT_ID / UNIT_ALIAS]")
+        println()
+        println("Example:   \"+APP_NAME+\" ColorableLight-8")
+        println("           \"+APP_NAME+\" 844a5b35-4b9c-4db2-9d22-4842db77bc95")
```

Comment:
Remove the backslash so that " is not escaped. Else the `APP_NAME` is not resolved. This can also be seen into your description of the pull request.